src/main/app/src/components/oppilaitos/OppilaitosinfoGrid.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import InfoGrid from '#/src/components/common/InfoGrid';
+import SchoolOutlinedIcon from '@material-ui/icons/SchoolOutlined';
+import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';
+import HomeWorkOutlinedIcon from '@material-ui/icons/HomeWorkOutlined';
+import PublicOutlinedIcon from '@material-ui/icons/PublicOutlined';
+import PeopleOutlineIcon from '@material-ui/icons/PeopleOutline';
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from '@material-ui/core';
+import { Localizer as l } from '#/src/tools/Utils';
+import _ from 'lodash';
+
+const useStyles = makeStyles((theme) => ({
+  koulutusInfoGridIcon: {
+    color: theme.palette.primary.main,
+  },
+}));
+
+const OppilaitosinfoGrid = (props) => {
+  const classes = useStyles();
+  const { opiskelijoita, osat, koulutusohjelmia, toimipisteita } = props;
+  const { t } = useTranslation();
+  const paikkakunnat = _.join(
+    _.sortBy(_.uniq(_.map(osat, `kotipaikka.nimi.${l.getLanguage()}`))),
+    ', '
+  );
+  const opetuskielet = _.join(
+    _.sortBy(
+      _.uniq(_.map(_.map(osat, `opetuskieli[0].nimi.${l.getLanguage()}`), _.capitalize))
+    ),
+    ', '
+  );
```

Comment:
En ole ehdotuksen koodia testannut, joten en ole varma onko täysin oikein, mutta nämä voi kirjoittaa huomattavasti luettavammin funktionaalisesti tähän tyyliin:

```suggestion
  const paikkakunnat = _fp.compose(
    _fp.join(', '),
    _fp.sortedUniq,
    _fp.uniq,
    _fp.map(_fp.capitalize),
    _fp.map(`kotipaikka.nimi.${l.getLanguage()}`)
  )(osat)
```
```